lib/Dialect/HW/HWOps.cpp

```diff
@@ -465,15 +467,28 @@ ModulePortInfo hw::getModulePortInfo(Operation *op) {
       type = inout.getElementType();
     }
 
+    StringAttr sym = {};
     auto direction = isInOut ? PortDirection::INOUT : PortDirection::INPUT;
-    inputs.push_back({argNames[i].cast<StringAttr>(), direction, type, i});
+    if (argAttrs && (i < argAttrs.size()))
+      if (auto s = argAttrs[i].cast<DictionaryAttr>())
+        if (auto symRef = s.get("hw.exportPort"))
+          sym = symRef.cast<FlatSymbolRefAttr>().getAttr();
```

Comment:
This pattern is replicated several times through this patch, can it be split out to a named helper function?

Reply:
Thanks, that makes sense, Moved this to a helper function to get the port symbols.